flax/nnx/transforms/iteration.py

```diff
@@ -714,6 +714,7 @@ def insert_index_mappings(x):
 
 def _scan_split_in(
   carry_deque: PytreeDeque[list[State | variablelib.Variable]],
+  carry_graphdefs_deque: PytreeDeque[graph.GraphDef | None],
```

Comment:
can we simplify this just having a `graphdef_queue: PytreeDeque[graph.GraphDef]` and pass all graphdefs through here?